### DISCUSSION TITLE
kata 2. No puedo continuar con el entorno virtual, por que me sale que no tengo instalado python y si lo tengo instalado

### DISCUSSION THREAD

**Discussion by SaulHernandezM** [LaunchX-InnovaccionVirtual/CursoIntroPython#215] [Q&A]
![Screenshot 2022-02-12 155133](https://user-images.githubusercontent.com/99091932/153736019-be92541e-73cd-4478-a19e-18435bdb1368.png)
![Screenshot 2022-02-12 200827](https://user-images.githubusercontent.com/99091932/153736025-600165a2-c6df-4eb4-9a92-fc9f08d84563.png)
Ya he intentado con todas las soluciones que dieron los demás miembros en otra discusión pero aun me sigue apareciendo que no engo instalado python, necesito ayuda ya no se que hacer 😭.

cosas que he intentado: python -m venv env, python3 -m venv env y todas las soluciones que le dieron a este miembro en su pregunta https://github.com/LaunchX-InnovaccionVirtual/CursoIntroPython/discussions/22#discussioncomment-2136771

**Comment by JshJLV** [comment]
Yo inicie mi entorno virtual con el comando " py -3 -m venv env "


De hecho yo seguí todos los pasos de aquí https://flask.palletsprojects.com/en/2.0.x/installation/#virtual-environments

**Reply by SaulHernandezM** [reply]
no me sale nada.

**Reply by JshJLV** [reply]
desde donde y como realizaste la instalacion de python

**Reply by SaulHernandezM** [reply]
desde el link que nos proporcionaron y también seleccione el path.

**Reply by JshJLV** [reply]
<img width="676" alt="captura_cmd" src="https://user-images.githubusercontent.com/90582121/153738657-df078ce7-db8f-4703-86e8-5563fde47895.png">

Mira, volví a repetir el proceso que hice, al inicio lo escribí con el comando que usaste en tu primera captura y me salió el mismo mensaje que te dio a ti

luego ya lo hice como el comando que te sugerí, la consola no te devuelve ningún mensaje, solo estas creando el entorno virtual asi que, si quieres comprobar si funciono, vete dentro de los archivos a la carpeta que creaste para tu entorno y debería salirte la carpeta recién creada
<img width="702" alt="captura_archivos" src="https://user-images.githubusercontent.com/90582121/153738937-bca8d7d3-3e1e-49c4-ab0b-f96d53122e22.png">
ahí esta la carpeta que cree, la que se llama env

De hecho después de escribir el comando la consola se queda como cargando, así podrías notar que se esta creando, que no te regrese ningún mensaje

ahí esta la carpeta que cree, la que se llama env

ya después escribo el comando de activación
env\Scripts\activate
nota que se usa la contrabarra, esta (\), no esta (/)

espero que así se pueda, suerte

**Reply by SaulHernandezM** [reply]
Gracias amigo si me funciono.

**Comment by doguedogue** [comment]
Prueba con

`$virtualenv env`

**Reply by SaulHernandezM** [reply]
![Screenshot 2022-02-12 211216](https://user-images.githubusercontent.com/99091932/153736664-e492ecee-4b66-4b85-9daa-3e0c2cbc2bbd.png)
Me salió esto.

**Reply by doguedogue** [reply]
Checa, esto le probó la compañera

https://github.com/LaunchX-InnovaccionVirtual/CursoIntroPython/discussions/93

**Comment by PlayerOneV** [comment]
Se pone python3 cuando tienes mas de una versión de python, prueba usando el comando sin el 3.
![image](https://user-images.githubusercontent.com/75816087/153739748-2926b968-8668-46ac-9144-e9d83ffeafff.png)

**Comment by charlyguz** [comment]
hola lo que pasa es que el código que esta escrito en la kata esta mal escrito y no lo reconoce en Windows mira prueba con esto:
>py -3 -m venv env
>env\Scripts\activate

**Comment by Laura-Karina-GR** [comment]
Hola, buenos días, también tengo problemas con esa kata, me sale esto :(
![image](https://user-images.githubusercontent.com/99380455/153759917-6fc3f037-ef52-4603-b4c4-d43e77edfdc2.png)

**Reply by JshJLV** [reply]
Lo que pasa es que tienes que poner " cd " (change directory) antes de escribir la ruta

En la primera ruta escribiste " cd/documents "

Pero en realidad sería " cd documents ", sin la / y con un espacio (sin las comillas claro)

Y así escribes las rutas como las necesites

Por ejemplo: cd documents/micurso/katas

**Reply by Laura-Karina-GR** [reply]
Va, deja lo intento, gracias

**Reply by Laura-Karina-GR** [reply]
Ya lo intenté y no me salió, mira
![image](https://user-images.githubusercontent.com/99380455/153761229-ed23f8c7-ebbe-48a4-95f9-e75ac2b9c1db.png)

**Reply by AndreAlmanza** [reply]
¡Hola Kari! podrías intentar usar el comando `dir`, este comando muestra en una lista las carpetas o archivos que hay en la ruta en la que te encuentras. Puede ser útil para buscar por qué no aparece tu carpeta documentos.

**Reply by AndreAlmanza** [reply]
Parece que el problema específico por el cual no te funciona el comando es porque estás utilizando `/` en lugar de `\`.
En windows para especificar una ruta debes separar las carpetas con `\`
ejemplo:
```
cd Documents\LaunchX
```
Inténtalo de esa forma y confírmame si se solucionó el problema.
![image](https://user-images.githubusercontent.com/48193837/153765330-bdbec492-663d-48b7-b1e8-d65374f4c616.png)
Esta es la combinación de teclas que me funcionan para colocar `\`

**Reply by FranciscoLL** [reply]
> Ya lo intenté y no me salió, mira ![image](https://user-images.githubusercontent.com/99380455/153761229-ed23f8c7-ebbe-48a4-95f9-e75ac2b9c1db.png)

Ya lo resolviste o aun tienes la duda?

**Reply by carlos-gmx3** [reply]
usa la diagonal invertida para las direcciones de los directorios cd \documents\Launch X...
la diagonal luego se usa como argumento en algunos comandos, como cuando vas a cambiar de unidad o disco
cd /D V:\PlayGround\Launch_x...
en este caso estoy cambiando a la unidad con la letra V:

**Comment by AndreAlmanza** [comment]
¡Hola, Saul! 
podrías enviar una captura de qué te aparece cuando colocas:
`python -m venv env`

**Comment by ide45** [comment]
Buenas, yo tuve el mismo problema y tal cómo dice ahí, instala python desde la store de microsotf. Eso me lo solucionó a mí

**Comment by carlos-gmx3** [comment]
Si despues de probar python --version aun no te sale nada es porque te falta agregar a python como variable de entorno en windows
simplemente para indicarle al sistema operativo que dispone de un interprete de python para manipular los programas escritos en este lenguaje de programación
https://tecnoloco.istocks.club/como-agregar-python-a-la-variable-path-de-windows-wiki-util/2020-10-14/
O como comentan, bien puedes instalar python desde la Microsoft store.
Teniendo ambas, VS Code te preguntará con cual interprete quieres ejecutar un archivo nuevo que tenga código Python
